.github/scripts/build-images.sh

```diff
@@ -0,0 +1,75 @@
+#!/bin/bash
+
+# @authors: Somya Arora, Arash Pourhabibi
+# @modified: Shanqing Lin
+
+
+# 1. Figure out the modified files
+if [ "${GITHUB_EVENT_NAME}" = "pull_request" ]; then
+  modified_files=$(git --no-pager diff --name-only ${PR_COMMIT_RANGE})
+else
+  modified_files=$(git --no-pager diff --name-only ${PUSH_COMMIT_RANGE})
+fi
+# 2.Get benchmark name and tag name for this build
+benchmark_name=${DH_REPO#*/}
+tag_name=$IMG_TAG
+if [ -z "$modified_files" ]; then
+    echo "No Modifications required."
+else
+    echo "Checking against modified files"
+fi
+# 3.Find out whether the files related with the current build were modified or not
+if (grep -q "$benchmark_name/$tag_name" <<<$modified_files) ||
+    (grep -q "$benchmark_name" <<<$modified_files && [ $tag_name = "latest" ]) ||
+    (grep -q "build-images.sh" <<<$modified_files) ||
+    (grep -q "build-images.yaml" <<<$modified_files) || 
+    ( [ $benchmark_name = "debian" ] && (grep -q "commons/base-os" <<<$modified_files) ); then
```

Comment:
Can't we make the base-os image naming convention the same as the other images so that you can reuse the `$benchmark_name/$tag_name` variables for this as well? Perhaps it makes sense to rename `benchmark_name` to something more general, like `image_repo`.

Reply:
I use `DF_PATH` to search the range. Some benchmarks like MySQL and dataset don't follow mapping between the folder structure and the image name. Using `DF_PATH` should solve this problem.